homeassistant/components/tplink/light.py

```diff
@@ -334,15 +357,15 @@ async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the light on."""
         brightness, transition = self._async_extract_brightness_transition(**kwargs)
         if ATTR_EFFECT in kwargs:
-            await self.device.set_effect(
+            await self._effect_module.set_effect(
                 kwargs[ATTR_EFFECT], brightness=brightness, transition=transition
             )
         elif ATTR_COLOR_TEMP_KELVIN in kwargs:
             if self.effect:
                 # If there is an effect in progress
-                # we have to set an HSV value to clear the effect
+                # we have to clear the effect
                 # before we can set a color temp
-                await self.device.set_hsv(0, 0, brightness)
+                await self._light_module.set_hsv(0, 0, brightness)
```

Comment:
It might also be possible to call `LightEffect.set_effect(LightEffect.LIGHT_EFFECTS_OFF) here but that needs to be tested with an `iot` lightstrip.

Reply:
Something that can be done in the `light` platform rewrite.